community/kernel/src/main/java/org/neo4j/kernel/impl/traversal/TraversalBranchImpl.java

```diff
@@ -174,7 +157,24 @@ protected TraversalBranch newNextBranch( Node node, Relationship relationship )
     @Override
     public void prune()
     {
-        relationships = PRUNED_ITERATOR;
+        closeRelationships();
+        relationships = Iterators.emptyResourceIterator();
+    }
+
+    @Override
+    public void close()
+    {
+        //TODO: not sure its a right place to close parent
+        parent.close();
```

Comment:
I reverted all the closing of branches and paths, in all places where it's unclear whenever I close them or not and add a flag to skip open statements checks for graph algo module